core/src/main/scala/org/apache/spark/memory/StoreUnifiedManager.scala

```diff
@@ -18,135 +18,85 @@ package org.apache.spark.memory
 }
 
 /**
-  * This class will store all the memory usage for GemFireXD boot up time when SparkEnv
-  * is not initialised.
-  * This class will not actually allocate any memory.
-  * It is just a temp account holder till SnappyUnifiedManager is started.
+  * A MemoryManager which simply delegates to configured Spark memory manager.
+  * This manager will be used in Connector mode. All SnappyData execution memory
+  * like encoder etc will account memory here.
   */
-class TempMemoryManager extends StoreUnifiedManager with Logging{
-
-  val memoryForObject = new mutable.HashMap[(String, MemoryMode), Long]()
-
-  override def acquireStorageMemoryForObject(objectName: String,
-      blockId: BlockId,
-      numBytes: Long,
-      memoryMode: MemoryMode,
-      buffer: UMMMemoryTracker,
-      shouldEvict: Boolean): Boolean = synchronized {
-    val key = objectName -> memoryMode
-    logDebug(s"Acquiring mem [TEMP] for $key $numBytes")
-    if (!memoryForObject.contains(key)) {
-      memoryForObject.put(key, 0L)
-    }
-    memoryForObject(key) += numBytes
-    true
-  }
-
-  override def dropStorageMemoryForObject(
-      objectName: String,
-      memoryMode: MemoryMode,
-      ignoreNumBytes : Long): Long = synchronized {
-    memoryForObject.remove(objectName -> memoryMode).getOrElse(0L)
-  }
-
-  override def releaseStorageMemoryForObject(
-      objectName: String,
-      numBytes: Long,
-      memoryMode: MemoryMode): Unit = synchronized {
-    memoryForObject(objectName -> memoryMode) -= numBytes
-  }
-
-  override def getStoragePoolMemoryUsed(memoryMode: MemoryMode): Long = 0L
-
-  override def getStoragePoolSize(memoryMode: MemoryMode): Long = 0L
-
-  override def getExecutionPoolUsedMemory(memoryMode: MemoryMode): Long = 0L
-
-  override def getExecutionPoolSize(memoryMode: MemoryMode): Long = 0L
-
-  override def getOffHeapMemory(objectName: String): Long = 0L
-
-  override def hasOffHeap: Boolean = false
-
-  override def logStats(): Unit = synchronized {
-    val memoryLog = new StringBuilder
-    val separator = "\n\t\t"
-    memoryLog.append("TempMemoryManager stats:")
-    for ((n, v) <- memoryForObject) {
-      memoryLog.append(separator).append(n).append(" = ").append(v)
-    }
-    logInfo(memoryLog.toString())
-  }
-
-  override def changeOffHeapOwnerToStorage(buffer: ByteBuffer,
-      allowNonAllocator: Boolean): Unit = {}
-}
-
-
-class NoOpSnappyMemoryManager extends StoreUnifiedManager with Logging {
+class DefaultMemoryManager extends StoreUnifiedManager with Logging {
 
   override def acquireStorageMemoryForObject(objectName: String,
       blockId: BlockId,
       numBytes: Long,
       memoryMode: MemoryMode,
       buffer: UMMMemoryTracker,
       shouldEvict: Boolean): Boolean = {
-    logDebug(s"Acquiring mem [NO-OP] for $objectName $numBytes")
-    true
+    logDebug(s"Acquiring DefaultManager meemory for $objectName $numBytes")
+    SparkEnv.get.memoryManager.acquireStorageMemory(blockId, numBytes, memoryMode)
```

Comment:
There might be some race conditions here that can cause NPE during startup/shutdown? Perhaps better to have SparkEnv.get.memoryManager as a val in the class.